core/model/objects/asset.cpp

```diff
@@ -13,41 +13,60 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-
 #include "asset.hpp"
+#include <iostream>
+
+namespace object {
+
+using Rule = json_parse::Rule;
+using Type = json_parse::Type;
+using Object = json_parse::Object;
+
+Asset::Asset(
+    Object obj
+):
+    domain(obj.dictSub["domain"].str),
+    name(obj.dictSub["name"].str),
+    value(obj.dictSub["value"].integer),
+    precision(obj.dictSub["precision"].integer)
+{}
+
+Asset::Asset(
+    const std::string& domain,
+    const std::string& name,
+    const unsigned long long& value,
+    const unsigned int& precision
+):
+    domain(domain),
+    name(name),
+    value(value),
+    precision(precision)
+{}
+
+Asset::Asset(
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Member variable 'Asset::precision' is not initialized in the constructor.](https://www.codacy.com/app/soramitsu/iroha/file/4478421695/issues/source?bid=3712501&fileBranchId=3923761#l46)